static/js/components/CommentTree.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import React from "react";
+import R from "ramda";
+import moment from "moment";
+
+import Card from "../components/Card";
+
+import type { Comment } from "../flow/discussionTypes";
+
+const renderTopLevelComment = (comment: Comment, idx: number) =>
+  <Card key={idx}>
+    <div className="top-level-comment">
+      {renderComment(0, comment)}
+    </div>
+  </Card>;
+
+const commentClassName = depth => depth > 0
+  ? "comment indented"
+  : "comment";
+
+const renderComment = R.curry((depth: number, comment: Comment) =>
+  <div className={commentClassName(depth)} key={`${depth}${comment.created}`}>
+    <div className="author-info">
+      <a className="author-name">
+        {comment.author_id}
+      </a>
+      <div>
+        {moment(comment.created).fromNow()}
+      </div>
+    </div>
+    <div>
+      {comment.text}
+    </div>
+    { R.map(renderComment(depth + 1), comment.replies) }
```

Comment:
I prefer the current approach I think, because I'd prefer for a comment which does not have any replies to not have an empty `.replies` div in it - in order to get around that I'd have to add another helper function or something which adds more complexity than what I have now, which is just a very simple recursive function.